presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -278,16 +287,23 @@ private ResultSet getNextResultSet()
         }
     }
 
-    private List<String> getMappedColumnNames(String minColumn, String maxColumn)
+    private List<String> getMappedColumnNames(Type dataType, String minColumn, String maxColumn)
     {
+        checkArgument((dataType == DATE) || (dataType == TIMESTAMP), "invalid data type");
```

Comment:
This is technically ok since these types are singletons, but types should be compared with `equals()` (and all the code does it that way)